web-api/src/ingestion/routes.rs

```diff
@@ -99,14 +100,29 @@ async fn new_documents(
     let (documents, mut failed_documents) = body
         .documents
         .into_iter()
-        .map(|document| match state.embedder.run(&document.snippet) {
-            Ok(embedding) => Ok((document, embedding)),
-            Err(err) => {
-                error!(
-                    "Document with id '{}' caused a PipelineError: {:#?}",
-                    document.id, err,
-                );
-                Err(document.id.into())
+        .map(|document| {
+            match state
+                .embedder
+                .run(&document.snippet)
+                .and_then(|res| res.normalized().map_err(InternalError::from_std))
```

Comment:
❓ wdyt about moving the call to `normalized()` into the `embedder.run()`? then we could guarantee that we never forget to normalize embeddings, independent of the call site.

Reply:
okido